### PR TITLE
LPS-110689 Adds Numpad keys to validInputKeyCodes list

### PR DESCRIPTION
**Pull request by diegonvs**
### Test Case

1. Global Menu Icon
2. Control Panel -> Configurations -> Adaptive Media
3. Add a new adaptive media(?) clicking on the plus button

Try to enter the width and height values by hitting the numbers above the letters on your keyboard: Works ✅ 
Try to enter the width and height values by hitting the numbers on the numberpad (num-lock on, of course): Doesn't work. ❌ 

Test Case additional considerations:

I wasn't unable to install a Linux distro to test it and It wasn't reproduced on Mac. However, running a Windows alongside a VM and using a Microsoft Numpad keyboard on a Firefox I was able to reproduce it.

-----

One interesting thing is because when using `key` instead of `keyCode` We don't need to care about Numpad different key codes :) (See https://github.com/liferay-frontend/liferay-portal/pull/153/commits/3bf070c9d322206ad601c907ec095004ce400e85)

See:
Screen Shot 2020-07-22 at 16 13 30
From: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

### CODE REVIEW

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by carloslancha** [comment]
Not an expert on this but [compatibility table for event.key](https://caniuse.com/#search=event.key) says that `event.key` on IE doesn't match completely the specification.

But also [compatibility table for event.keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) says its deprecated, so... 😅 

I'd just check if it's working as expected in IE

**Comment by diegonvs** [comment]
ci:stop

**Comment by diegonvs** [comment]
As far I know, old browsers like IE11 can return different `.key` from the specification. See https://github.com/liferay/clay/issues/3430#issuecomment-653112978

However, on React's source code, We can see a normalisation for these different keys. See https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/react-dom/src/events/getEventKey.js#L12-L29

Also, I forget adding 'Up' and 'Down' for 'ArrowUp' and 'ArrowDown' keys compatibility 😴  😂 

Updated on: 57de87a

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by carloslancha** [comment]
image

**Comment by diegonvs** [comment]
Once relevant is done and greener, I'll forward

**Comment by jbalsas** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-web/src/main/resources/META-INF/resources/adaptive_media/js/EditAdaptiveMediaConfig.es.js [hunk 1]
```diff
@@ -129,9 +126,7 @@ class EditAdaptiveMediaConfig extends PortletBase {
 	 * @param {KeyboardEvent} event The keyboard event.
 	 */
 	handleKeyDown_(event) {
-		const code = event.keyCode || event.charCode;
-
-		if (this.validInputKeyCodes_.indexOf(code) == -1) {
+		if (VALID_INPUT_KEYS.indexOf(event.key) == -1) {
```
Comment (wincent) [inline modules/apps/adaptive-media/adaptive-media-web/src/main/resources/META-INF/resources/adaptive_media/js/EditAdaptiveMediaConfig.es.js:129]: Prefer `===` over `==`.

As a matter of habit, if you want to look up something by value, you should be looking it up in a data structure that has `O(1)` lookup.

So, maybe you should change `VALID_INPUT_KEYS` to be `new Set([...])`, and then you can do:

```js
if (VALID_INPUT_KEYS.has(event.key)) {
    // ....
}
```

(I know you are just carrying forward the existing `indexOf`, but may as well improve things while we're at it as per the Boy Scout Rule.)

Comment (diegonvs) [reply modules/apps/adaptive-media/adaptive-media-web/src/main/resources/META-INF/resources/adaptive_media/js/EditAdaptiveMediaConfig.es.js:129]: I was caught again in the law of the good scout.

Fixing!

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by wincent** [comment]
lg2m: @diegonvs — I force-pushed to your branch because I know Brian is going to want [the keys sorted](https://github.com/liferay-frontend/liferay-portal/pull/153/commits/917adab620d11c0fe9c3538d89125b667a36f4fb). Let's see if we can forward this now.

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**